<?xml version="1.0" encoding="UTF-8"?>
<definitions id="definitions" 
  xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL"
  xmlns:camunda="http://camunda.org/schema/1.0/bpmn"
  targetNamespace="Examples">

  <error id="myError" errorCode="MyError" />
  <error id="myException" errorCode="org.camunda.bpm.engine.test.bpmn.event.error.MyBusinessException" />

  <process id="testProcess" isExecutable="true">
  
    <startEvent id="start" />

    <sequenceFlow id="flow1" sourceRef="start" targetRef="serviceTask" />
    
    <serviceTask id="serviceTask" camunda:expression="${true}" />

    <boundaryEvent id="catchException" attachedToRef="serviceTask">
      <errorEventDefinition errorRef="myException"/>
    </boundaryEvent>

    <sequenceFlow id="flow2" sourceRef="serviceTask" targetRef="serviceTask2" />
    
    <serviceTask id="serviceTask2" camunda:class="org.camunda.bpm.engine.test.bpmn.event.error.ThrowErrorDelegate" />
    
    <sequenceFlow id="flow7" sourceRef="serviceTask2" targetRef="end" />

    <endEvent id="end" />

    <sequenceFlow id="flow5" sourceRef="catchException" targetRef="userTaskException" />

    <userTask id="userTaskException" name="Exception Task" />

    <sequenceFlow id="flow6" sourceRef="userTaskException" targetRef="endException" />

    <endEvent id="endException" />

  </process>

</definitions>
